src/core/lib/security/transport/secure_endpoint.cc

```diff
@@ -270,11 +334,16 @@ static void endpoint_read(grpc_endpoint* secure_ep, grpc_slice_buffer* slices,
 }
 
 static void flush_write_staging_buffer(secure_endpoint* ep, uint8_t** cur,
```

Comment:
similarly, this whole function needs to be guarded by write_mu

Reply:
Done